src/components/ui/button.svelte

```diff
@@ -34,6 +27,18 @@
     classname ? classname : 'btn',
   ].filter(v => v !== false);
 
+  const getRippleColor = () => {
+    if (ripple) {
+      if (isFilled) {
+        return 'rgba(255, 255, 255, .35)';
+      } else {
+        if (isDanger) return 'rgba(186, 3, 3, .25)';
+        else return 'rgba(56, 120, 0, .25)';
+      }
+    }
+    return null;
```

Comment:
Same comment here: using guard clauses can make it only one level of nesting
```suggestion
    if(!ripple)
      return null;
    if(isFilled)
      return 'rgba(255, 255, 255, 0.35)';
    if(isDanger)
      return 'rgba(186, 3, 3, 0.25)';
    return 'rgba(56, 120, 0, 0.25)';
```

Reply:
Also, the function is used in a line before it is declared (using `const`!).
And it can be inlined using a reactive block instead.
```js
$: {
  if(!ripple)
    rippleColor = null;
  // ...
}
```